src/stdune/exn_with_backtrace.mli

```diff
@@ -0,0 +1,21 @@
+
+(** An exception together with the backtrace that raised it. *)
+
+type t = {
+  exn : exn;
+  backtrace : Printexc.raw_backtrace;
+}
+
+val try_with : ('a -> 'b) -> 'a -> ('b, t) Result.t
```

Comment:
It's a bit of a shame that `Result.try_with` has a slightly different type:

```
val try_with : (unit -> 'a) -> ('a, t) Result.t
```

Should we stick to just one convention? (I don't have a preference for either one)

Reply:
Changed to the one from `Result.try_with`, given that that's used in non-zero number of places.